docs: add troubleshooting resource requests (#2001)
kubernetes-sigs/kueue: site/content/en/docs/tasks/troubleshooting/troubleshooting_jobs.md

   job-my-job-19797   user-queue   cluster-queue   9m45s
   ```

> ## Does my cluster-queue match the job resource requests?

Reviewer: I would put this as a `###` after the `Pending workload` section.

> ## Does my cluster-queue match the job resource requests?

Reviewer: ```suggestion
## Does my ClusterQueue have the resource requests that the job requires?
```

When you submit a job that has a resource request, for example:

```bash

> $ kubectl get jobs job-0-9-size-6 -o json | jq -r .spec.template.spec.containers[0].resources

Reviewer: ```suggestion
$ kubectl get jobs job-0-9-size-6 -o jsonpath='.spec.template.spec.containers[0].resources'
```

Although this wouldn't pretty print

Author: I do like the pretty printing <3 But happy to change it if the docs already have a convention.

```
```console
{
  "limits": {
    "cpu": "2"
  },
  "requests": {
    "cpu": "2"
  }
}
```

> If your cluster queue does not have a definition for the _requests_ the job cannot be admitted. For the job above you would want to ensure that you've defined "cpu" under `resourceGroups`, as shown below:

Reviewer: ```suggestion
If your ClusterQueue does not have a definition for the _requests_, Kueue cannot admit the job. For the job above, you would want to ensure that you've defined `cpu` under `resourceGroups`, as shown below:
```

Reviewer: Although, I would try to avoid repeating too many things that are already described somewhere else. You can link to https://kueue.sigs.k8s.io/docs/concepts/cluster_queue/#resource-groups

The idea of this document is to give a decision tree for debugging, and link to the relevant documentation for the specifics.

```yaml
apiVersion: kueue.x-k8s.io/v1beta1
kind: ClusterQueue
metadata:
  name: "cluster-queue"
spec:
  namespaceSelector: {}
  resourceGroups:
  - coveredResources: ["cpu"]
    flavors:
    - name: "default-flavor"
      resources:
      - name: "cpu"
        nominalQuota: 40
```

Generally speaking, if you find that workloads are not admitted, a good debugging strategy is to check that you've covered resources that are requested, and then carefully describe each of the job and workload, along with looking at logs for kueue. You can typically find a log or condition message that gives a hint to why the queue is not moving.

> 

Reviewer: this is already explained in the other sections.

Author: I am going to say here that redundancy is OK, especially in documentation. I don't think we want to be maintaining equivalent information (e.g., some spec or definition) in two places, but reminders about "here is a good way to debug" are always good to have multiple times. The user reader very likely has not seen/read all the "the other sections."

Author: I did wind up removing it, resolving!

    type: QuotaReserved
```

### Does my ClusterQueue have the resource requests that the job requires?

When you submit a job that has a resource request, for example:

```bash
$ kubectl get jobs job-0-9-size-6 -o json | jq -r .spec.template.spec.containers[0].resources
```
```console
{
  "limits": {
    "cpu": "2"
  },
  "requests": {
    "cpu": "2"
  }
}
```

> If your ClusterQueue does not have a definition for the _requests_, Kueue cannot admit the job. For the job above, you would want to ensure that you've defined `cpu` under `resourceGroups`, as shown below:

Reviewer: ```suggestion
If your ClusterQueue does not have a definition for the `requests`, Kueue cannot admit the job. For the job above, you should define `cpu` quotas under `resourceGroups`. A ClusterQueue defining `cpu` quota looks like the following:
```
